components/emoji/emoji.test.js

```diff
@@ -1,183 +1,135 @@
 import { createLocalVue, mount } from '@vue/test-utils';
 import DtEmoji from './emoji.vue';
 import { setEmojiAssetUrlSmall, setEmojiAssetUrlLarge, setCustomEmojiUrl, setCustomEmojiJson } from '@/common/emoji.js';
-import { flushPromises } from '@/common/utils.js';
 import customEmojiJson from '@/common/custom-emoji.json';
 
 setEmojiAssetUrlSmall('https://mockstorage.com/emojis/', '.png');
 setEmojiAssetUrlLarge('https://mockstorage.com/emojis/', '.svg');
 setCustomEmojiUrl('https://mockstorage.com/emojis/');
 
-// Constants
-const basePropsData = {};
+const MOCK_EXPECTED_SMILE_SRC = 'https://mockstorage.com/emojis/1f604.svg';
+const MOCK_EXPECTED_SMILE_SRC_SMALL = 'https://mockstorage.com/emojis/1f604.png';
+const MOCK_EXPECTED_LAUGHING_SRC = 'https://mockstorage.com/emojis/1f606.svg';
+const MOCK_EXPECTED_POINT_UP_LIGHT = 'https://mockstorage.com/emojis/261d-1f3fb.svg';
+const MOCK_EXPECTED_SHIP_IT = 'https://mockstorage.com/emojis/shipit.png';
 
-describe('DtEmoji Tests', () => {
-  let testContext;
+const baseProps = { code: '' };
```

Comment:
Sth I need to point here.

The code prop in `emoji.vue` the default value is null, if we provide a null value its broke the component.

In our method: 
```
getEmojiData () {
      this.emojiData = codeToEmojiData(this.code);
    },
```
We have this `codeToEmojiData(this.code);` which is doing this:
`if (code.startsWith(':') && code.endsWith(':')) {`

So if you send a null , this `code.startsWith(':')` brokes the component, so we can change the default to an empty string `''` or as required prop or maybe we can safe check this and return an empty string...

Reply:
hmm I'm thinking it would make sense for code to be a required prop.

Reply:
I was thinking required prop is my preferred option but this could be a breaking change...
Maybe `null` to empty string `''` would work also...

Could we move forward with this as `required` prop?

Reply:
hmm do you really think it would break anything? Would anyone really be using emoji without the code prop?

Reply:
dunno, just asking, I will update it here then